mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,384 @@
+   :language: aspx-cs
+   :lines: 2-4
+   :emphasize-lines: 2,3
+  
+4. Run the app. You can use your favorite browser to inspect the source and verify that the markup has changed as promised.
+
+The ``[TargetElement]`` attribute above only targets HTML markup that provides an attribute value name of "bold". The ``<bold>`` element was not modified by the tag helper.  
+
+5. Comment out the ``[TargetElement]`` attribute line and it will default to targeting ``<bold>`` tags, that is, HTML markup of the form ``<bold>``. Remember, the default naming convention will match the class name **Bold**\TagHelper to ``<bold>`` tags.
+
+6. Run the app and verify that the ``<bold>`` tag is processed by the tag helper.
+
+Decorating a class with multiple ``TargetElement`` statements results in a logical-OR of the targets. For example, using the code below, a bold tag or a bold attribute will match.
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/zBoldTagHelperCopy.cs
+   :language: c#
+   :lines: 5-6
+ 
+When multiple attributes are added to the same statement, the runtime treats them as a logical-AND. For example, in the code below, an HTML element must contain both a bold tag and a bold attribute to match.
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/zBoldTagHelperCopy2.cs
+   :language: c#
+   :lines: 5
+
+For a good example of a bootstrap progress bar that targets a tag and an attribute, see `Creating custom MVC 6 Tag Helpers <http://www.davepaquette.com/archive/2015/06/22/creating-custom-mvc-6-tag-helpers.aspx>`__.
+
+You can also use the ``TargetElement`` to change the name of the targeted element. For example if you wanted the ``BoldTagHelper`` to target ``<MyBold>`` tags, you would use the following attribute:
+
+.. code-block:: c#
+
+	    [TargetElement("MyBold")]
+
+Web site information Tag Helper
+------------------------------------
+
+#. Add a *Models* folder.
+#. Add the following ``WebsiteContext`` class to the *Models* folder:
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/Models/WebsiteContext.cs
+    :language: c#
+
+3. Add the following ``WebsiteInformationTagHelper`` class to the *TagHelpers* folder.
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/WebsiteInformationTagHelper.cs
+    :language: c#
+
+Notes: 
+
+- As mentioned previously, the ASP.NET 5 runtime translates Pascal-cased C# class names and properties for tag helpers into `lower kebab case <http://c2.com/cgi/wiki?KebabCase>`__. Therefore, to use the ``WebsiteInformationTagHelper`` in Razor, you'll write ``<website-information />``.
+- We are not explicitly identifying the target element with the ``TargetElement`` attribute, so the default of ``website-information`` will be targeted. If you applied the following attribute (note it's not kebab case but matches the class name):
```

Comment:
[attribute], yes it's much better with []. That violates our official style guide. Thank goodness we're not bound to that.